pkg/cloud/cloud.go

```diff
@@ -283,29 +283,37 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 	)
 	capacityGiB := util.BytesToGiB(diskOptions.CapacityBytes)
 
+	if diskOptions.IOPS > 0 && diskOptions.IOPSPerGB > 0 {
+		return nil, fmt.Errorf("invalid StorageClass parameters; specify either IOPS or IOPSPerGb, not both")
+	}
+
 	switch diskOptions.VolumeType {
 	case VolumeTypeGP2, VolumeTypeSC1, VolumeTypeST1, VolumeTypeSBG1, VolumeTypeSBP1, VolumeTypeStandard:
 		createType = diskOptions.VolumeType
 	case VolumeTypeIO1:
 		createType = diskOptions.VolumeType
-		iops, err = capIOPS(diskOptions.VolumeType, capacityGiB, int64(diskOptions.IOPSPerGB), io1MinTotalIOPS, io1MaxTotalIOPS, io1MaxIOPSPerGB, diskOptions.AllowIOPSPerGBIncrease)
-		if err != nil {
-			return nil, err
+		if diskOptions.IOPS > 0 {
+			iops = int64(diskOptions.IOPS)
+		} else {
+			iops, err = capIOPS(diskOptions.VolumeType, capacityGiB, int64(diskOptions.IOPSPerGB), io1MinTotalIOPS, io1MaxTotalIOPS, io1MaxIOPSPerGB, diskOptions.AllowIOPSPerGBIncrease)
+			if err != nil {
+				return nil, err
+			}
 		}
 	case VolumeTypeIO2:
 		createType = diskOptions.VolumeType
-		iops, err = capIOPS(diskOptions.VolumeType, capacityGiB, int64(diskOptions.IOPSPerGB), io2MinTotalIOPS, io2MaxTotalIOPS, io2MaxIOPSPerGB, diskOptions.AllowIOPSPerGBIncrease)
-		if err != nil {
-			return nil, err
+		if diskOptions.IOPS > 0 {
+			iops = int64(diskOptions.IOPS)
+		} else {
+			iops, err = capIOPS(diskOptions.VolumeType, capacityGiB, int64(diskOptions.IOPSPerGB), io2MinTotalIOPS, io2MaxTotalIOPS, io2MaxIOPSPerGB, diskOptions.AllowIOPSPerGBIncrease)
+			if err != nil {
+				return nil, err
+			}
 		}
-	case VolumeTypeGP3:
+	case VolumeTypeGP3, "":
 		createType = diskOptions.VolumeType
 		iops = int64(diskOptions.IOPS)
 		throughput = int64(diskOptions.Throughput)
-	case "":
-		createType = DefaultVolumeType
```

Comment:
If you're going to remove this line you should also remove the `DefaultVolumeType` constant

Reply:
Done.